File: lc_go/lcGo/slidingWindows/findMaxAverage.go
package main

import "math"

func findMaxAverage(nums []int, k int) (ans float64) {

	s := math.MinInt

	cs := 0

	for i, x := range nums {
		cs += x

		if i < k-1 {
			continue
		}

		s = max(cs, s)

		cs -= nums[i-k+1]
	}
	return float64(s) / float64(k)
}

func main() {

}

File: lc_cpp/lcCpp/SlidingWindows/findMaxAverage.cpp
//
// Created by 24276 on 2025/7/26.
//


#include <limits.h>
#include <vector>
class find_max_average {
public:
    double findMaxAverage(std::vector<int>& nums, int k) {
        int ans = INT_MIN;
        int sum = 0;

        for (int i = 0;i < k; i++) {
            sum += nums[i];
        }

        ans = sum;

        for (int i = k; i < nums.size(); i++) {
            sum += nums[i] - nums[i - k];
            ans = std::max(ans, sum);
        }

        return static_cast<double>(ans) / k;
        }
};